### ISSUE TITLE
Symfony 5.2 breaks JMS cache

### ISSUE THREAD

**Issue by bobvandevijver** [schmittjoh/JMSSerializerBundle#823]
After upgrading to Symfony 5.2, this bundle breaks with the following error:

```
The directory "/var/www/app/var/cache/dev/jms_serializer" does not exist.
```

This only happens when `bin/console cache:clear` is run twice, as it seems that it no longer runs the warmup stage on the second invocation. I also opened a report with Symfony, so we can decide what can be done to correctly fix this.

See the Symfony issue for more information: https://github.com/symfony/symfony/issues/39232

**Comment by mbabker** [comment]
Just to note here, this is only true if you're using the default filesystem cache configuration (which, is pretty bad).

Inside one of my apps, we changed the `jms_serializer.metadata.cache` config node from its default "file" setting (which uses that path by default) to a custom service using the `Metadata\Cache\PsrCacheAdapter` class and a cache pool from Symfony's Cache component, and we aren't running into this issue since the cache is being managed by Symfony instead of the `Metadata\Cache\FileCache` class as is the default (whose constructor is throwing this error because it expects the directory to exist and has no fallback to try and create it).

```yaml
# config/services.yaml
services:
    app.jms_serializer.metadata.cache:
        class: Metadata\Cache\PsrCacheAdapter
        arguments:
            - '%env(CACHE_PREFIX)%_jms_metadata'
            - '@cache.jms_serializer.metadata'

# config/packages/cache.yaml
framework:
    cache:
        pools:
            cache.jms_serializer.metadata:
                adapter: cache.adapter.system

# config/packages/jms_serializer.yaml
jms_serializer:
    metadata:
        cache: 'app.jms_serializer.metadata.cache'
```

**Comment by bobvandevijver** [comment]
This should be solved in Symfony 5.2.1 and higher (thanks to https://github.com/symfony/symfony/pull/39360).

**Comment by ruudk** [comment]
@bobvandevijver Thanks for the followup. I will wait for the new release :)
